interoperator/pkg/controller/sfserviceinstance/sfserviceinstance_controller.go

```diff
@@ -0,0 +1,333 @@
+		scheme:         mgr.GetScheme(),
+		clusterFactory: clusterFactory,
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("serviceinstance-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to ServiceInstance
+	err = c.Watch(&source.Kind{Type: &osbv1alpha1.SFServiceInstance{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	postgres := &unstructured.Unstructured{}
+	postgres.SetKind("Postgres")
+	postgres.SetAPIVersion("kubedb.com/v1alpha1")
+	postgres2 := &unstructured.Unstructured{}
+	postgres2.SetKind("Postgresql")
+	postgres2.SetAPIVersion("kubernetes.sapcloud.io/v1alpha1")
+	director := &unstructured.Unstructured{}
+	director.SetKind("Director")
+	director.SetAPIVersion("deployment.servicefabrik.io/v1alpha1")
+	docker := &unstructured.Unstructured{}
+	docker.SetKind("Docker")
+	docker.SetAPIVersion("deployment.servicefabrik.io/v1alpha1")
+	postgresqlmts := &unstructured.Unstructured{}
+	postgresqlmts.SetKind("PostgresqlMT")
+	postgresqlmts.SetAPIVersion("deployment.servicefabrik.io/v1alpha1")
+	vhostmts := &unstructured.Unstructured{}
+	vhostmts.SetKind("VirtualHost")
+	vhostmts.SetAPIVersion("deployment.servicefabrik.io/v1alpha1")
+	subresources := []runtime.Object{
+		&appsv1.Deployment{},
+		&corev1.ConfigMap{},
+		postgres,
+		postgres2,
+		director,
+		docker,
+		postgresqlmts,
+		vhostmts,
+	}
+
+	for _, subresource := range subresources {
+		err = c.Watch(&source.Kind{Type: subresource}, &handler.EnqueueRequestForOwner{
+			IsController: true,
+			OwnerType:    &osbv1alpha1.SFServiceInstance{},
+		})
+		if err != nil {
+			log.Printf("%v", err)
+		}
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileServiceInstance{}
+
+// ReconcileServiceInstance reconciles a ServiceInstance object
+type ReconcileServiceInstance struct {
+	client.Client
+	scheme         *runtime.Scheme
+	clusterFactory *clusterFactory.ClusterFactory
+}
+
+// Reconcile reads that state of the cluster for a ServiceInstance object and makes changes based on the state read
+// and what is in the ServiceInstance.Spec
+// TODO(user): Modify this Reconcile function to implement your Controller logic.  The scaffolding writes
+// a Deployment as an example
+// Automatically generate RBAC rules to allow the Controller to read and write Deployments
+// +kubebuilder:rbac:groups=kubedb.com,resources=Postgres,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=kubernetes.sapcloud.io,resources=postgresql,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=deployment.servicefabrik.io,resources=director,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=deployment.servicefabrik.io,resources=docker,verbs=get;list;watch;create;update;patch;delete
```

Comment:
If not required, please remove.

Reply:
These are necessary in order to generate correct auto generated code.